source/core/shape.cc

```diff
@@ -392,22 +400,28 @@ template <int dim>
 std::shared_ptr<Shape<dim>>
 CutHollowSphere<dim>::static_copy() const
 {
-  std::shared_ptr<Shape<dim>> copy = std::make_shared<CutHollowSphere<dim>>(
-    radius, cut_depth, shell_thickness, this->position, this->orientation);
+  std::shared_ptr<Shape<dim>> copy =
+    std::make_shared<CutHollowSphere<dim>>(this->radius,
+                                           this->cut_depth,
+                                           this->shell_thickness,
+                                           this->position,
+                                           this->orientation);
   return copy;
 }
 
 template <int dim>
 double
-CutHollowSphere<dim>::displaced_volume(const double /*fluid_density*/)
-{
-  std::cout
-    << "Warning: For a cut hollow sphere, the real volume will be lower than "
-       "output."
-    << std::endl;
+CutHollowSphere<dim>::displaced_volume(
+  const double /*fluid_density*/,
+  std::shared_ptr<ConditionalOStream> pcout)
+{
+  if (pcout)
+    *pcout
+      << "Warning: For a cut hollow sphere, the real volume will be lower than "
+      << "output." << std::endl;
```

Comment:
What is meant by this here? I don't think the displaced volume part is a good place to print stuff... Only print outside...